### PR TITLE
z_actor Dialog System Documentation

### PR DESCRIPTION
**Pull request by kleril**
Documenting the (mostly) unused monolithic dialog system handler that lives in z_actor.c
The Deku Tree Sprout is the only remaining thing in the codebase that uses this codepath

### CODE REVIEW

File: src/code/z_actor.c [hunk 1]
```diff
@@ -5644,107 +5672,138 @@ s32 func_800374E0(PlayState* play, Actor* actor, u16 textId) {
...
 }
 
-s32 func_80037CB8(PlayState* play, Actor* actor, s16 arg2) {
+/**
+ * Processes the current message state
+ * Note: Only used by Deku Tree Sprout, see an individual actor's overlay .c file for where their dialog logic actually resides
+ *
+ * @param play current game state
+ * @param actor the actor we're trying to talk to
+ * @param actorID (s16) the ID of the actor we're trying to talk to
```
Comment (Dragorn421) [inline src/code/z_actor.c:5792]:
```suggestion
 * @param actor the actor we're trying to talk to
 * @param actorID the ID of the actor we're trying to talk to
```
Style:
We just omit `@param` for obvious arguments like play which is ubiquitous
and we don't indicate types

"we" here to me should refer to the function user, that is the actor calling `Actor_TalkUpdate`
but if you say "the actor we're trying to talk to" then who is "we" ? the player actor?

File: src/code/z_actor.c [hunk 2]
```diff
@@ -5644,107 +5672,138 @@ s32 func_800374E0(PlayState* play, Actor* actor, u16 textId) {
...
+/**
+ * Returns a text ID value for an actor given the current set of world flags
+ * Note: Only used by Deku Tree Sprout, see an individual actor's overlay .c file for where their dialog logic actually resides
+ *
+ * @param play current game state
+ * @param actor the actor we're trying to talk to
+ * @param pad (s32) unused
+ * @return 0 if we found and loaded a response, 1 if something has gone wrong
+ */
+s32 Actor_ProcessDialogChoice(PlayState* play, Actor* actor, s32 pad) {
```
Comment (Dragorn421) [inline src/code/z_actor.c:5782]:
```suggestion
s32 Actor_ProcessDialogChoice(PlayState* play, Actor* actor, s32 unused) {
```
"pad" is used out of habit/semi-convention for variable declarations where it's needed to "pad" the stack for matching, but it doesn't make sense to use this name for arguments.
We use `unused` elsewhere

**Review by Dragorn421 — COMMENTED** [review]
This system looks like a copy of `Npc_UpdateTalking` except `Npc_UpdateTalking` takes callbacks instead of hardcoding things into a giant switch.
Hard to say which of the two came first if they weren't written at the same time

Specifically, `Npc_UpdateTalking` (callbacks version) corresponds to this PR's `Actor_TalkUpdate` (giant switch version).
- `Npc_UpdateTalking` : `Actor_TalkUpdate`
- `NpcGetTextIdFunc getTextId` : `Actor_GetTextID`
- `NpcUpdateTalkStateFunc updateTalkState` : `Actor_ProcessMessage`

It could be a good idea to name things similarly between the two
Also it sounds like this could use the `Npc_` prefix instead of generic `Actor_`, or possibly some other prefix that would contain this "lamost unused dialog system", I am thinking of `NpcImpl_` "npc implementation (instead of callbacks)" but idk

**Comment by fig02** [comment]
Closing due to inactivity. 
Apologies for personally not reviewing this PR quick enough. I wasnt a fan of the proposed `NpcImpl_` name but never figured out an alternative to suggest.

Feel free to reopen this PR if you get back around to oot things and want to maintain the PR again

**Pull request closed without merging**